.github/workflows/terraform-tests.yml

```
@@ -0,0 +1,107 @@
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+name: Terraform Tests
+jobs:
+
+  prepare_list:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - id: set-matrix
+        run: echo "::set-output name=matrix::$(python tests/terraform/get-tf-services.py --names)"
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
+  test_service:
+    needs: prepare_list
+    strategy:
+      fail-fast: false
+      matrix:
+        service: ${{ fromJson(needs.prepare_list.outputs.matrix) }}
+        go-version: [1.18.x]
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      DNS_ADDRESS: 0
+      AWS_DEFAULT_REGION: us-east-2
+      AWS_ALTERNATE_REGION: eu-west-1
+
+    steps:
+    - uses: actions/setup-go@v3
+      with:
+        go-version: ${{ matrix.go-version }}
+
+    - name: Checkout Localstack
+      uses: actions/checkout@v3
+
+    - name: Checkout Terraform AWS Provider
+      uses: actions/checkout@v3
+      with:
+        repository: hashicorp/terraform-provider-aws
+        path: './terraform-provider-aws'
+
+    - name: Set up Python 3.10.5
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.10.5'
+
+    - name: Caching Python .venv
+      id: python-cache
+      uses: actions/cache@v2
+      with:
+        path: ./.venv/
+        key: ${{ runner.os }}-venv-${{ hashFiles('**/setup.cfg') }}
+        restore-keys: |
+          ${{ runner.os }}-venv-
+
+    - name: Upgrading Pip
```

nit: Maybe we could rename this to "Install system dependencies"? It does not only upgrade `pip`, but installs `libsasl2-dev` and `pyyaml`.

```suggestion
    - name: Install system dependencies
```

---

Updated